File: CSharp/Chess/Chess/Piece.cs
using System;

namespace Chess
{
	public abstract class Piece
	{
		protected char character;
		private Color color;
		private Position position;
		
		public Piece(Color color, Position position)
		{
			this.color = color;
			this.position = position;
		}
		
		public abstract string GetName();
		public abstract int GetMaxCount();
		
		public override string ToString()
		{
			return color + " " + GetName() + " (" + position + ")";
		}
	}
}

File: Ruby/Chess/piece.rb

require './position.rb'

class Piece
  
  def initialize(color, position)
    @name = "Piece"
    @color = color
    @position = position
  end
  
  def GetName()
    return @name
  end
  
  def to_s
    return @color + " " + GetName() + " (" + @position.to_s() + ")"
  end
  
end
